Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -36,33 +36,566 @@ class Resources:
+            azure_datalake_analytics_catalog_and_job_endpoint=suffixes.get(
+                'azure_datalake_analytics_catalog_and_job_endpoint',
+                defaults.suffixes.azure_datalake_analytics_catalog_and_job_endpoint),
+            acr_login_server_endpoint=suffixes.get('acr_login_server_endpoint', defaults.suffixes.acr_login_server_endpoint),
+            synapse_analytics_endpoint=suffixes.get('synapse_analytics_endpoint', defaults.suffixes.synapse_analytics_endpoint),
+            attestation_endpoint=suffixes.get('attestation_endpoint', defaults.suffixes.attestation_endpoint),
+        ))
+
+
+def microsoft_defender_for_endpoint_get_base_url(endpoint_type, url, is_gcc=None):
+    # Backward compatible argument parsing, preserve the url and is_gcc functionality if provided, otherwise use endpoint_type.
+    log_message_append = ""
+    if is_gcc:  # Backward compatible.
+        endpoint_type = "US GCC"
+        log_message_append = f" ,Overriding endpoint to {endpoint_type}, backward compatible."
+    elif endpoint_type == MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE_CUSTOM or not endpoint_type:
+        # When the integration was configured before our Azure Cloud support, the value will be None.
+        if not url:
+            if endpoint_type == MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE_CUSTOM:
+                raise DemistoException("Endpoint type is set to 'Custom' but no URL was provided.")
+            raise DemistoException("'Endpoint Type' is not set and no URL was provided.")
+    endpoint_type = MICROSOFT_DEFENDER_FOR_ENDPOINT_TYPE.get(endpoint_type, 'com')
+    url = url or MICROSOFT_DEFENDER_FOR_ENDPOINT_API[endpoint_type]
+    demisto.info(f"Using url:{url}, endpoint type:{endpoint_type}{log_message_append}")
+    return endpoint_type, url
+
+
+def get_azure_cloud(params, integration_name):
+    azure_cloud_arg = params.get('azure_cloud')
+    if not azure_cloud_arg or azure_cloud_arg == AZURE_CLOUD_NAME_CUSTOM:
+        # Backward compatibility before the azure cloud settings.
+        if 'server_url' in params:
+            return create_custom_azure_cloud(integration_name, defaults=AZURE_WORLDWIDE_CLOUD,
+                                             endpoints={'resource_manager': params['server_url']})
+        if 'azure_ad_endpoint' in params:
+            return create_custom_azure_cloud(integration_name, defaults=AZURE_WORLDWIDE_CLOUD,
+                                             endpoints={
+                                                 'active_directory': params.get('azure_ad_endpoint',
+                                                                                'https://login.microsoftonline.com')
+                                             })
+
+    # There is no need for backward compatibility support, as the integration didn't support it to begin with.
+    return AZURE_CLOUDS[azure_cloud_arg]
```

Comment:
```suggestion
    return AZURE_CLOUDS.get(azure_cloud_arg, AZURE_WORLDWIDE_CLOUD)
```
  in case like in AzureKeyVault the azure_cloud_arg can be '' without server_url or azure_ad_endpoint in params
```suggestion
    return AZURE_CLOUDS[azure_cloud_arg]
```